count/include/count_min_impl.hpp

```diff
@@ -0,0 +1,234 @@
+#ifndef COUNT_MIN_IMPL_HPP_
+#define COUNT_MIN_IMPL_HPP_
+
+#include "MurmurHash3.h"
+#include <random>
+
+namespace datasketches {
+
+template<typename T>
+count_min_sketch<T>::count_min_sketch(uint64_t num_hashes, uint64_t num_buckets, uint64_t seed):
+num_hashes(num_hashes),
+num_buckets(num_buckets),
+seed(seed){
+  sketch_length = num_hashes*num_buckets ;
+  sketch.resize(sketch_length) ;
+  assert(num_buckets >= 3); // need relative error epsilon at most 1.
```

Comment:
we have a policy to throw exceptions instead of asserts. std::invalid_argument in this case I suppose